src/Microsoft.ML.Data/Transforms/KeyToValue.cs

```diff
@@ -568,9 +568,9 @@ public void SaveAsOnnx(OnnxContext ctx)
 
                     if (!ctx.ContainsColumn(inputColumnName))
                         continue;
-
+                    string srcVariableName = ctx.GetVariableName(inputColumnName);
                     var dstVariableName = ctx.AddIntermediateVariable(_types[iinfo], info.outputColumnName, true);
-                    if (!_kvMaps[iinfo].SaveOnnx(ctx, inputColumnName, dstVariableName))
+                    if (!_kvMaps[iinfo].SaveOnnx(ctx, srcVariableName, dstVariableName))
```

Comment:
I think it would be a good idea to add a test case specifically because we don't have tests for this. Also please verify this fixes the Nimbus issue.

Reply:
Sounds good. I restructured the tests and got rid of a KeyToValue test that seems redundant and is not formatted to test different types of values.